README.md

```diff
@@ -1,46 +1,214 @@
+Golemrpc allows users to make task requests without having to install Golem client and take care of port forwarding, GNT purchase etc. Library handles user defined tasks by communicating with Golem over RPC (remote procedure call) interface. This Golem node will be requesting tasks to P2P network on user's behalf. Requesting node can be hosted by anyone, anywhere e.g. on AWS EC2 instance.
+
+Having a Golem node running on localhost is also supported, but by default this library will handle input/output files transfer over TCP. To disable it and take advantage of sharing the filesystem with Golem node use `remote` argument in `RPCComponent` class.
+
+## Installation
+
+Use package installer for Python3.6:
 
 ```sh
-# First node listening on port 61000
-python $GOLEM_DIR/golemapp.py --datadir=/home/$USER/datadir1 --password=node1 --accept-terms --rpc-address=localhost:61000
-# Second node listening on port 61001 and pointing to first as it's peer
-python $GOLEM_DIR/golemapp.py --datadir=/home/$USER/datadir2 --peer=localhost:40102 --rpc-address=localhost:61001
+pip3 install golemrpc
 ```
 
-Go to `golemrpc/examples/` directory and run one of the examples:
+# User guide
 
-```sh
-source venv/bin/activate
-cd golemrpc/examples
-python raspa_task.py
+## Basic example
+
+First set up a logging module::
+
+```python
+import logging
+logging.basicConfig(level=logging.INFO)
+```
+
+Then create and start an RPCComponent:
+
+```python
+from golemrpc.rpccomponent import RPCComponent
+rpc = RPCComponent(
+    host='35.158.100.160',
+    port=61000,
+    cli_secret_filepath='golemcli_aws.tck',
+    rpc_cert_filepath='rpc_cert_aws.pem'
+)
+rpc.start()
+```
+
+It will spawn a separate thread that will handle communication with Golem on `35.158.100.160:61000`. Now define a `dummy_task` function 
+to compute on Golem provider's machine:
+
+```python
+def dummy_task(args):
+    return 1 + args['b']
+```
+
+Create a golemrpc `CreateTask` message with task type `GLambda` (an abbreviation for Golem Lambda, a type of task that takes serialized Python function and dictionary as input and sends them to the provider for computation. For more information about messages and events see [this](https://github.com/golemfactory/golemrpc/blob/threaded/docs/messages.md)):
+
+```python
+message = {
+    'type': 'CreateTask',
+    'task': {
+        'type': 'GLambda',
+        'options': {
+            'method': dummy_task,
+            'args': {'b': 2},
+        },
+    }
+}
+```
+
+Send it to RPCComponent:
+
+```python
+rpc.post(message)
+```
+
+Now poll the RPC message queue for new events. First expected event is `TaskCreatedEvent`:
+
+```python
+task_created_event = rpc.poll(timeout=None)
+```
+
+It contains `task_id`assigned to this particular task by the requesting Golem node. One can use it to keep track of tasks in case there are more than one being computed at a time. Second message coming from RPCComponent should be `TaskResults` containing filepaths to actual results:
```

Comment:
```suggestion
It contains `task_id` assigned to this particular task by the requesting Golem node. One can use it to keep track of tasks in case there are more than one being computed at any one time. Second message coming from the RPCComponent should be `TaskResults` containing file paths to the actual results:
```